[BUG] UnboundLocalError: local variable 'channelsID' referenced before assignment

### Describe the bug
Error `UnboundLocalError: local variable 'channelsID' referenced before assignment` when `pytest -k "prepare_data_for_ci"` is called as part of GaNDLF workflow run from OpenFL.

Check [these steps](https://github.com/securefederatedai/openfl/blob/develop/.github/workflows/gandlf.yml#L67).

```
01: Constructing training CSVs
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.10.17/x64/lib/python3.10/site-packages/_pytest/main.py", line 281, in wrap_session
INTERNALERROR>     config.hook.pytest_sessionstart(session=session)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.10.17/x64/lib/python3.10/site-packages/pluggy/_hooks.py", line 513, in __call__
INTERNALERROR>     return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.10.17/x64/lib/python3.10/site-packages/pluggy/_manager.py", line 120, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.10.17/x64/lib/python3.10/site-packages/pluggy/_callers.py", line 139, in _multicall
INTERNALERROR>     raise exception.with_traceback(exception.__traceback__)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.10.17/x64/lib/python3.10/site-packages/pluggy/_callers.py", line 122, in _multicall
INTERNALERROR>     teardown.throw(exception)  # type: ignore[union-attr]
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.10.17/x64/lib/python3.10/site-packages/_pytest/logging.py", line 782, in pytest_sessionstart
INTERNALERROR>     return (yield)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.10.17/x64/lib/python3.10/site-packages/pluggy/_callers.py", line 103, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/home/runner/work/openfl/openfl/gandlf/testing/conftest.py", line 50, in pytest_sessionstart
INTERNALERROR>     prerequisites_constructTrainingCSV()
INTERNALERROR>   File "/home/runner/work/openfl/openfl/gandlf/testing/test_full.py", line 170, in prerequisites_constructTrainingCSV
INTERNALERROR>     channelsID,
INTERNALERROR> UnboundLocalError: local variable 'channelsID' referenced before assignment
```

Right now, it is a blocker for us, as it's part of our PR pipeline.

**NOTE**: Version being used is 0.1.1

### To Reproduce
Steps to reproduce the behavior:
1. Install GaNDLF repo with 0.1.1 tag
2. Perform these steps - https://github.com/securefederatedai/openfl/blob/develop/.github/workflows/gandlf.yml#L59
3. When `test_prepare_data_for_ci` is called, the `UnboundLocalError` comes.

Looks like none of the if/else blocks are satisfied and that's when it fails - https://github.com/mlcommons/GaNDLF/blob/0.1.1/testing/test_full.py#L170

A failed job for your reference - https://github.com/securefederatedai/openfl/actions/runs/14892023481/job/41826836616#step:7:584

### Expected behavior
No error

@sarthakpati - With OpenFL pointing to master branch of GaNDLF (and the master being my forked repo where I am printing the value of inputDir and application), I am ending up with below error.

```
ImportError while loading conftest '/home/runner/work/openfl/openfl/gandlf/testing/conftest.py'.
testing/conftest.py:6: in <module>
    from .test_full import (
testing/test_full.py:15: in <module>
    from GANDLF.data.ImagesFromDataFrame import ImagesFromDataFrame
...
/opt/hostedtoolcache/Python/3.10.17/x64/lib/python3.10/ctypes/__init__.py:374: in __init__
    self._handle = _dlopen(self._name, mode)
E   OSError: /opt/hostedtoolcache/Python/3.10.17/x64/lib/python3.10/site-packages/torchaudio/lib/libtorchaudio.so: undefined symbol: _ZN3c1010Dispatcher17runRecordFunctionERN2at14RecordFunctionESt17reference_wrapperIKNS_14FunctionSchemaEENS_11DispatchKeyE
```

Full error here - https://github.com/noopurintel/openfl/actions/runs/14908469949/job/41876542683#step:7:675

My forked branch with just print statements - https://github.com/mlcommons/GaNDLF/compare/master...noopurintel:GaNDLF:master

--- a/testing/test_full.py
+++ b/testing/test_full.py
@@ -196,7 +196,9 @@ def prerequisites_constructTrainingCSV():
             with open(
                 inputDir + "/train_" + application_data + ".csv", "r"
             ) as read_f, open(
-                inputDir + "/train_" + application_data_regression + ".csv", "w", newline=""
+                inputDir + "/train_" + application_data_regression + ".csv",
+                "w",
+                newline="",
             ) as write_reg, open(
                 inputDir + "/train_" + application_data_classification + ".csv",
                 "w",

--- a/testing/test_full.py
+++ b/testing/test_full.py
@@ -152,73 +152,75 @@ def prerequisites_constructTrainingCSV():
             os.remove(os.path.join(inputDir, item))
 
     for application_data in os.listdir(inputDir):
-        currentApplicationDir = os.path.join(inputDir, application_data)
+        if "segmentation" in application_data:
+            # this is to ensure other types of unit-testing data do not inadvertently get pulled in during testing
+            currentApplicationDir = os.path.join(inputDir, application_data)
 
-        if "2d_rad_segmentation" in application_data:
-            channelsID = "image.png"
-            labelID = "mask.png"
-        elif "3d_rad_segmentation" in application_data:
-            channelsID = "image"
-            labelID = "mask"
-        elif "2d_histo_segmentation" in application_data:
-            channelsID = "image"
-            labelID = "mask"
-        # else:
-        #     continue
-        outputFile = inputDir + "/train_" + application_data + ".csv"
-        outputFile_rel = inputDir + "/train_" + application_data + "_relative.csv"
-        # Test with various combinations of relative/absolute paths
-        # Absolute input/output
-        writeTrainingCSV(
-            currentApplicationDir,
-            channelsID,
-            labelID,
-            outputFile,
-            relativizePathsToOutput=False,
-        )
-        writeTrainingCSV(
-            currentApplicationDir,
-            channelsID,
-            labelID,
-            outputFile_rel,
-            relativizePathsToOutput=True,
-        )
+            if "2d_rad_segmentation" in application_data:
+                channelsID = "image.png"
+                labelID = "mask.png"
+            elif "3d_rad_segmentation" in application_data:
+                channelsID = "image"
+                labelID = "mask"
+            elif "2d_histo_segmentation" in application_data:
+                channelsID = "image"
+                labelID = "mask"
+            # else:
+            #     continue
+            outputFile = inputDir + "/train_" + application_data + ".csv"
+            outputFile_rel = inputDir + "/train_" + application_data + "_relative.csv"
+            # Test with various combinations of relative/absolute paths
+            # Absolute input/output
+            writeTrainingCSV(
+                currentApplicationDir,
+                channelsID,
+                labelID,
+                outputFile,
+                relativizePathsToOutput=False,
+            )
+            writeTrainingCSV(
+                currentApplicationDir,
+                channelsID,
+                labelID,
+                outputFile_rel,
+                relativizePathsToOutput=True,
+            )
 
-        # write regression and classification files
-        application_data_regression = application_data.replace(
-            "segmentation", "regression"
-        )
-        application_data_classification = application_data.replace(
-            "segmentation", "classification"
-        )
-        with open(
-            inputDir + "/train_" + application_data + ".csv", "r"
-        ) as read_f, open(
-            inputDir + "/train_" + application_data_regression + ".csv", "w", newline=""
-        ) as write_reg, open(
-            inputDir + "/train_" + application_data_classification + ".csv",
-            "w",
-            newline="",
-        ) as write_class:
-            csv_reader = csv.reader(read_f)
-            csv_writer_1 = csv.writer(write_reg)
-            csv_writer_2 = csv.writer(write_class)
-            i = 0
-            for row in csv_reader:
-                if i == 0:
-                    row.append("ValueToPredict")
-                    csv_writer_2.writerow(row)
-                    # row.append('ValueToPredict_2')
-                    csv_writer_1.writerow(row)
-                else:
-                    row_regression = copy.deepcopy(row)
-                    row_classification = copy.deepcopy(row)
-                    row_regression.append(str(random.uniform(0, 1)))
-                    # row_regression.append(str(random.uniform(0, 1)))
-                    row_classification.append(str(random.randint(0, 2)))
-                    csv_writer_1.writerow(row_regression)
-                    csv_writer_2.writerow(row_classification)
-                i += 1
+            # write regression and classification files
+            application_data_regression = application_data.replace(
+                "segmentation", "regression"
+            )
+            application_data_classification = application_data.replace(
+                "segmentation", "classification"
+            )
+            with open(
+                inputDir + "/train_" + application_data + ".csv", "r"
+            ) as read_f, open(
+                inputDir + "/train_" + application_data_regression + ".csv", "w", newline=""
+            ) as write_reg, open(
+                inputDir + "/train_" + application_data_classification + ".csv",
+                "w",
+                newline="",
+            ) as write_class:
+                csv_reader = csv.reader(read_f)
+                csv_writer_1 = csv.writer(write_reg)
+                csv_writer_2 = csv.writer(write_class)
+                i = 0
+                for row in csv_reader:
+                    if i == 0:
+                        row.append("ValueToPredict")
+                        csv_writer_2.writerow(row)
+                        # row.append('ValueToPredict_2')
+                        csv_writer_1.writerow(row)
+                    else:
+                        row_regression = copy.deepcopy(row)
+                        row_classification = copy.deepcopy(row)
+                        row_regression.append(str(random.uniform(0, 1)))
+                        # row_regression.append(str(random.uniform(0, 1)))
+                        row_classification.append(str(random.randint(0, 2)))
+                        csv_writer_1.writerow(row_regression)
+                        csv_writer_2.writerow(row_classification)
+                    i += 1
 
 
 def test_prepare_data_for_ci():

--- a/testing/test_full.py
+++ b/testing/test_full.py
@@ -152,77 +152,76 @@ def prerequisites_constructTrainingCSV():
             os.remove(os.path.join(inputDir, item))
 
     for application_data in os.listdir(inputDir):
-        if "segmentation" in application_data:
-            # this is to ensure other types of unit-testing data do not inadvertently get pulled in during testing
-            currentApplicationDir = os.path.join(inputDir, application_data)
+        if "segmentation" not in application_data:
+            continue
+        # this is to ensure other types of unit-testing data do not inadvertently get pulled in during testing
+        currentApplicationDir = os.path.join(inputDir, application_data)
 
-            if "2d_rad_segmentation" in application_data:
-                channelsID = "image.png"
-                labelID = "mask.png"
-            elif "3d_rad_segmentation" in application_data:
-                channelsID = "image"
-                labelID = "mask"
-            elif "2d_histo_segmentation" in application_data:
-                channelsID = "image"
-                labelID = "mask"
-            # else:
-            #     continue
-            outputFile = inputDir + "/train_" + application_data + ".csv"
-            outputFile_rel = inputDir + "/train_" + application_data + "_relative.csv"
-            # Test with various combinations of relative/absolute paths
-            # Absolute input/output
-            writeTrainingCSV(
-                currentApplicationDir,
-                channelsID,
-                labelID,
-                outputFile,
-                relativizePathsToOutput=False,
-            )
-            writeTrainingCSV(
-                currentApplicationDir,
-                channelsID,
-                labelID,
-                outputFile_rel,
-                relativizePathsToOutput=True,
-            )
+        if "2d_rad_segmentation" in application_data:
+            channelsID = "image.png"
+            labelID = "mask.png"
+        elif "3d_rad_segmentation" in application_data:
+            channelsID = "image"
+            labelID = "mask"
+        elif "2d_histo_segmentation" in application_data:
+            channelsID = "image"
+            labelID = "mask"
+        # else:
+        #     continue
+        outputFile = inputDir + "/train_" + application_data + ".csv"
+        outputFile_rel = inputDir + "/train_" + application_data + "_relative.csv"
+        # Test with various combinations of relative/absolute paths
+        # Absolute input/output
+        writeTrainingCSV(
+            currentApplicationDir,
+            channelsID,
+            labelID,
+            outputFile,
+            relativizePathsToOutput=False,
+        )
+        writeTrainingCSV(
+            currentApplicationDir,
+            channelsID,
+            labelID,
+            outputFile_rel,
+            relativizePathsToOutput=True,
+        )
 
-            # write regression and classification files
-            application_data_regression = application_data.replace(
-                "segmentation", "regression"
-            )
-            application_data_classification = application_data.replace(
-                "segmentation", "classification"
-            )
-            with open(
-                inputDir + "/train_" + application_data + ".csv", "r"
-            ) as read_f, open(
-                inputDir + "/train_" + application_data_regression + ".csv",
-                "w",
-                newline="",
-            ) as write_reg, open(
-                inputDir + "/train_" + application_data_classification + ".csv",
-                "w",
-                newline="",
-            ) as write_class:
-                csv_reader = csv.reader(read_f)
-                csv_writer_1 = csv.writer(write_reg)
-                csv_writer_2 = csv.writer(write_class)
-                i = 0
-                for row in csv_reader:
-                    if i == 0:
-                        row.append("ValueToPredict")
-                        csv_writer_2.writerow(row)
-                        # row.append('ValueToPredict_2')
-                        csv_writer_1.writerow(row)
-                    else:
-                        row_regression = copy.deepcopy(row)
-                        row_classification = copy.deepcopy(row)
-                        row_regression.append(str(random.uniform(0, 1)))
-                        # row_regression.append(str(random.uniform(0, 1)))
-                        row_classification.append(str(random.randint(0, 2)))
-                        csv_writer_1.writerow(row_regression)
-                        csv_writer_2.writerow(row_classification)
-                    i += 1
+        # write regression and classification files
+        application_data_regression = application_data.replace(
+            "segmentation", "regression"
+        )
+        application_data_classification = application_data.replace(
+            "segmentation", "classification"
+        )
+        with open(
+            inputDir + "/train_" + application_data + ".csv", "r"
+        ) as read_f, open(
+            inputDir + "/train_" + application_data_regression + ".csv", "w", newline=""
+        ) as write_reg, open(
+            inputDir + "/train_" + application_data_classification + ".csv",
+            "w",
+            newline="",
+        ) as write_class:
+            csv_reader = csv.reader(read_f)
+            csv_writer_1 = csv.writer(write_reg)
+            csv_writer_2 = csv.writer(write_class)
+            i = 0
+            for row in csv_reader:
+                if i == 0:
+                    row.append("ValueToPredict")
+                    csv_writer_2.writerow(row)
+                    # row.append('ValueToPredict_2')
+                    csv_writer_1.writerow(row)
+                else:
+                    row_regression = copy.deepcopy(row)
+                    row_classification = copy.deepcopy(row)
+                    row_regression.append(str(random.uniform(0, 1)))
+                    # row_regression.append(str(random.uniform(0, 1)))
+                    row_classification.append(str(random.randint(0, 2)))
+                    csv_writer_1.writerow(row_regression)
+                    csv_writer_2.writerow(row_classification)
+                i += 1
 
 
 def test_prepare_data_for_ci():

